### PR TITLE
Class decorate

### PR DESCRIPTION
**Pull request by getssh**
Hello, I have worked on implementing the decorator

- Added Nameable class as a top-level class
- Implemented a method called correct_name that will raise a NotImplementedError
- Made Person class inherits from Nameable
- Created base class that inherits from Nameable
- Implemented the correct_name method to base class
- Added CapitalizeDecorator to capitalize the given correct_name
- Added TrimmerDecorator to trim long string
- Added person object in person.rb file to test and check the outputs
- checked syntax errors with rubocop

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -21,3 +28,35 @@ def of_age?
     false
   end
 end
+
+class PersonDecorator < Nameable
```
Comment (t-yanick) [inline person.rb:32]: - [x] You need to add the `attr_accessor` method in order to be able to  access and set instance variable `nameable` from outside the class. 
```ruby
class Decorator < Nameable
    attr_accessor :nameable
```

**Review by t-yanick — CHANGES_REQUESTED** [review]
# SATATUS: CHANGES REQUIRED

Hi @getssh , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: All Decorators added as per project requirements
:heavy_check_mark: Linter check passing

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (getssh) [reply person.rb:32]: Thank you for the review @t-yanick, I have added the attr_accessor as recommended. Thank you

**Review by TadesseDev — APPROVED** [review]
### Hi @getssh 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**